### PR TITLE
[FEA] Physics informed examples using the `PhysicsInformer` utility

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Updates some of the key examples to use the newer `PhysicsInformer` utility. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: examples/cfd/ldc_pinns/config.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    in_channels: 2
+    dimension: 2
+    latent_channels: 32
+    fno_layers: 4
+    fno_modes: 12
+    padding: 9
+
+scheduler:
+  initial_lr: 1.E-3
+  decay_rate: .9995
```
Comment (RishikeshRanade) [inline examples/cfd/ldc_pinns/config.yaml:33]: Can the type of gradient calculation used be part of the config?

File: examples/cfd/stokes_mgn/pi_fine_tuning_gnn.py [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+        v = torch.zeros_like(y)
+        return u, v
+
+    def loss(self):
+        out = self.model(
+            self.dgl_graph.ndata["x"], self.dgl_graph.edata["x"], self.dgl_graph
+        )
+
+        # inflow points
+        mask_inflow = (
```
Comment (RishikeshRanade) [inline examples/cfd/stokes_mgn/pi_fine_tuning_gnn.py:204]: Is this the masking because we don't calculate the gradients on boundary elements/voxels/nodes? Should this be taken care of when the gradient is returned by the PhysicsInformer utility instead of the user having to code it up?

**Review by RishikeshRanade — APPROVED** [review]

File: examples/cfd/darcy_physics_informed/conf/config_deeponet.yaml [hunk 3]
```diff
@@ -24,7 +24,7 @@ start_lr: 0.001
 gamma: 0.99948708
 max_epochs: 50
 
-phy_wt: 0.1
+physics_wt: 0.1
```
Comment (Alexey-Kamenev) [inline examples/cfd/darcy_physics_informed/conf/config_deeponet.yaml:27]: I wouldn't shorten `weight` either :)

File: examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py [hunk 4]
```diff
@@ -110,6 +107,14 @@ def main(cfg: DictConfig):
         padding=cfg.model.fno.padding,
     ).to(device)
 
+    phy_informer = PhysicsInformer(
+        required_outputs=["diffusion_u"],
+        equations=darcy,
+        grad_method="finite_difference",
+        device=device,
+        fd_dx=1 / 240,
```
Comment (Alexey-Kamenev) [inline examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py:115]: Is this a known constant? Maybe name it or add comment?

**Review by Alexey-Kamenev — APPROVED** [review]
LGTM!

File: examples/cfd/ldc_pinns/config.yaml (hunk 1 above)
Comment (ktangsali) [reply examples/cfd/ldc_pinns/config.yaml:33]: It is possible. But since this is an introductory example, I decided not to do that. The idea was to provide a minimal code with a focus on the utilities. For FD and Least Squares, the dataloader pipeline will have to be different and enabling that via config can become a bit verbose.

File: examples/cfd/stokes_mgn/pi_fine_tuning_gnn.py (hunk 2 above)
Comment (ktangsali) [reply examples/cfd/stokes_mgn/pi_fine_tuning_gnn.py:204]: So actually the `PhysicsInformer` utility will compute residuals on any points that the user passes. Since here, the points passed include the boundary points, the residuals computed on those points are not required as we need to apply BCs on them instead. Hence the masking was added. Potentially we have a higher level of abstraction at a later point?

File: examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py (hunk 4 above)
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py]: Done :)
